robot/Cumulus/tests/browser/engagement_plans/add_engagement_plan_to_contact.robot

```diff
@@ -2,34 +2,72 @@
 
 Resource        robot/Cumulus/resources/NPSP.robot
 Library         cumulusci.robotframework.PageObjects
+...             robot/Cumulus/resources/ContactPageObject.py
+...             robot/Cumulus/resources/AccountPageObject.py
 ...             robot/Cumulus/resources/EngagementPlanTemplatesPageObject.py
-Suite Setup     Open Test Browser
+...             robot/Cumulus/resources/NPSP.py
+
+Suite Setup     Run keywords
+...             Open Test Browser
+...             Setup Test Data
 Suite Teardown  Delete Records and Close Browser
 
-*** Test Cases ***
+***Keywords***
+# Setup a contact with parameters specified
+Setup Test Data
+    Setupdata   contact   contact_data=${contact_fields}
+
+*** Variables ***
+&{contact_fields}  Email=test@example.com
 
+*** Test Cases ***
 Create a Contact and Add Engagement Plan
-    [tags]  unstable
-    ${plan_name}     ${task1_1}    ${sub_task1_1}     ${task2_1}     Create Engagement Plan
-    Set Suite Variable    ${plan_name}
-    &{contact} =  API Create Contact    MailingStreet=50 Fremont Street    MailingCity=San Francisco    MailingPostalCode=95320    MailingState=CA    MailingCountry=USA
-    Store Session Record    Account    &{contact}[AccountId]
-    Go To Record Home  &{contact}[Id]
-    Select Tab  Related
-    Click Related List Button  Engagement Plans    New
-    Populate Lookup Field    Engagement Plan Template    ${plan_name}
-    Click Modal Button        Save
-    Validate Related Record Count    Engagement Plans   1
-
-Delete Engagement Plan
-    [tags]  unstable
-    ${plan_num}    Verify Eng Plan Exists    ${plan_name}
-    Click Related Item Popup Link    Engagement Plans    ${plan_num}    Delete
-    Click Modal Button        Delete
-    Verify Occurence          Engagement Plans       0
-    
-Verify Tasks Exist Under Activity
-    [tags]  unstable
-    Scroll Page To Location    0    0
-    Click More Activity Button    
-    Check Activity Tasks    ${task1}    ${sub_task}    ${task2}
+    [Documentation]                      Create an Engagement plan Template with two tasks and one subtask. And  a contact
+    ...                                  Link the  engagement plan for the contact and verify
+    ...                                  There is one engagement plan set up for the contact
+    [tags]                               W-038641                 feature:Engagements
+
+
+    Go To Page                                       Listing                            Engagement_Plan_Template__c
+    Go To Engagement Plan Page                       create
+    Wait For Locator                                 frame                              Manage Engagement Plan Template
+
+    Select Frame And Click Element                   Manage Engagement Plan Template    id    idName
```

Comment:
waiting for a frame and selecting frame should be hidden inside the keyword as a user doesn't need to know about those and its invisible

Reply:
Took care and hid the finer details in a keyword